### PR TITLE
Change how we publish public AMIs

### PR DESCRIPTION
**Pull request by brikis98**
Our AMI publishing script seemed to be broken (https://github.com/hashicorp/terraform-aws-consul/issues/76). I was struggling to figure out why and realized that Packer can now do most of what that script does with the `ami_regions` and `ami_groups` parameters. I made those changes... and then realized that the actual issue was that we were publishing the AMIs to the wrong account (to Phx DevOps rather than our public AMIs account). Doh.

Overall, I think this solution is simpler and better overall, so perhaps my work has not gone to waste.

### CODE REVIEW

File: _ci/publish-amis.sh [hunk 1]
```diff
@@ -7,51 +7,35 @@
...
+echo "Building Packer template $packer_template_path (builder: $builder_name) and sharing it with all AWS accounts in the following regions: $all_aws_regions"
+
+# Copying AMIs to many regions can take longer than Packer's default wait timeouts, so we increase them here per
+# https://github.com/hashicorp/packer/issues/6536
+export AWS_MAX_ATTEMPTS=240
+export AWS_POLL_DELAY_SECONDS=15
+
+# We publish the AMIs to a different AWS account, so set those credentials
+export AWS_ACCESS_KEY_ID="$PUBLISH_AMI_AWS_ACCESS_KEY_ID"
+export AWS_SECRET_ACCESS_KEY="$PUBLISH_AMI_AWS_SECRET_ACCESS_KEY"
```
Comment (brikis98) [inline _ci/publish-amis.sh]: The two lines above are the actual bug fix. The rest is just simplifying how we publish AMIs.

File: circle.yml [hunk 2]
```diff
@@ -27,13 +27,5 @@ deployment:
...
-    branch: publish-amis
-    commands:
-      # We generally only want to build AMIs on new releases, but when we are setting up AMIs in a new account for the
-      # first time, we want to build the AMIs but NOT run automated tests, since those tests will fail without an existing
-      # AMI already in the AWS Account.
-      - ~/$CIRCLE_PROJECT_REPONAME/_ci/publish-amis.sh "ubuntu16-ami"
-      - ~/$CIRCLE_PROJECT_REPONAME/_ci/publish-amis.sh "amazon-linux-ami"
\ No newline at end of file
+      - ~/$CIRCLE_PROJECT_REPONAME/_ci/publish-amis.sh "/home/ubuntu/$CIRCLE_PROJECT_REPONAME/examples/consul-ami/consul.json" "ubuntu16-ami"
+      - ~/$CIRCLE_PROJECT_REPONAME/_ci/publish-amis.sh "/home/ubuntu/$CIRCLE_PROJECT_REPONAME/examples/consul-ami/consul.json" "amazon-linux-ami"
```
Comment (Etiene) [inline circle.yml]: I'll have to update the circle PR with those 👍

File: examples/consul-ami/consul.json [hunk 3]
```diff
@@ -2,7 +2,9 @@
   "min_packer_version": "0.12.0",
   "variables": {
     "aws_region": "us-east-1",
-    "consul_version": "1.2.2",
+    "consul_version": "1.0.0",
```
Comment (Etiene) [inline examples/consul-ami/consul.json]: You dowgraded the default version?

File: circle.yml (hunk 2 above)
Comment (brikis98) [reply circle.yml]: Yes plz!

File: examples/consul-ami/consul.json (hunk 3 above)
Comment (brikis98) [reply examples/consul-ami/consul.json]: Whoops. Unintentional, will fix. Good catch 👍

**Comment by brikis98** [comment]
Test failure is due to lack of a published AMI! Merging now and will create a new release to check this all works.

**Pull request merged**